core/object/object.cpp

```diff
@@ -1303,10 +1303,7 @@ void Object::get_signals_connected_to_this(List<Connection> *p_connections) cons
 }
 
 Error Object::connect(const StringName &p_signal, const Callable &p_callable, uint32_t p_flags) {
-	ERR_FAIL_COND_V_MSG(p_callable.is_null(), ERR_INVALID_PARAMETER, "Cannot connect to '" + p_signal + "': the provided callable is null.");
-
-	Object *target_object = p_callable.get_object();
-	ERR_FAIL_NULL_V_MSG(target_object, ERR_INVALID_PARAMETER, "Cannot connect to '" + p_signal + "' to callable '" + p_callable + "': the callable object is null.");
+	ERR_FAIL_COND_V_MSG(!p_callable.is_valid(), ERR_INVALID_PARAMETER, "Cannot connect to '" + p_signal + "': the provided callable is not valid: " + p_callable);
```

Comment:
`is_valid()` may be too strong here. If you look at the changes in `editor/plugins` you can see there are callables being created in a constructor that refer to `this`. It seems during the constructor the object isn't fully registered, meaning these callables aren't yet valid, and will be rejected by this line here.

I'm not sure how I like connecting invalid callables, but it was previous behavior to do so, so I'll probably change this line back to use `is_null()`, rather than try to chase down all the not-yet-valid-callables in Godot.

Reply:
> so I'll probably change this line back to use `is_null()`

That makes sense to me! And then the two changes to use `callable_mp()` where we weren't before can be removed.

Reply:
Just updated the PR for this. I opted for the hybrid approach I mentioned in our discussion on RocketChat, as its actually the most minimal change here (perseveres all original checks). Also removed the `callable_mp()` changes.

Reply:
Thanks! This looks good to me :-)